compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -2646,6 +2651,13 @@ public boolean isReAtomNode(NodeKind kind) {
         }
     }
 
+    private boolean isXmlnsInDifferentCompUnit(BXMLNSSymbol symbol, BXMLNSSymbol foundSym) {
+        if (foundSym.compUnit != null && symbol.compUnit != null) {
```

Comment:
I would extract the `sym.compUnit != null` check out to something like `isModuleXmlns`.